airflow/www/utils.py

```diff
@@ -779,6 +787,72 @@ class UtcAwareFilterConverter(fab_sqlafilters.SQLAFilterConverter):
     """Retrieve conversion tables for UTC-Aware filters."""
 
 
+class XComFilterStartsWith(fab_sqlafilters.FilterStartsWith):
+    """Starts With filter for XCom values."""
+
+    def apply(self, query: Query, value: str) -> Query:
+        query, trimmed_value = get_trimmed_field(query, self.model, self.column_name)
+        return query.filter(trimmed_value.ilike(value + "%"))
```

Comment:
follow up of the "another idea" above

```suggestion
    def apply(self, query: Query, value: str) -> Query:
        return generate_filter_value_query(
            query=query,
            model=self.model,
            column_name=self.column_name, 
            filter_cond=lambda trimmed_value: trimmed_value.ilike(f"%{value}")
        )
```